### PR TITLE
feat(dapp) use timbre for logging

### PR DESCRIPTION
**Pull request by billkube**
# Description

updating debug info to use timbre logging 

Closes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

# Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation website
- [ ] I have made corresponding changes to the literal docs
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by drew-patel — APPROVED** [review]
Beautiful. Thank you!

**Comment by crimeminister** [comment]
Hey @billkube, you might want to use the `trace` or `debug` logging level for some of these logging calls, depending on how permanent they are. And for what it's worth, I usually like to log maps or objects as many logging systems, e.g. StackDriver will do useful things with logged data.

File: dapp/src/dapp/components/login.cljs [hunk 1]
```diff
@@ -17,18 +17,18 @@
...
-  (prn "open the modal")
+  (log/trace "open the modal")
   (let [modal (Web3Modal. (clj->js provider-options))]
     ;(.clearCachedProvider modal)
     (-> (.connect modal)
         ;; TODO: figure out why this won't re-prompt wallet if password was not entered at prompt
          (.then (fn [provider]
                   ;; dispatch the provider
-                  (prn "provider")
+                  (log/debug {:provider "provider"})
```
Comment (drew-patel) [inline dapp/src/dapp/components/login.cljs:27]: This only shows the `"provider"` string. Did you want to replace this with just `provider` so we can see the object?

File: dapp/src/dapp/components/login.cljs [hunk 2]
```diff
@@ -17,18 +17,18 @@
...
                   (re-frame/dispatch [::wallet/web3-modal provider])))
          (.catch (fn [error]
                    (.clearCachedProvider modal)
-                   (js/console.log error))))))
+                   (log/error error))))))
```
Comment (drew-patel) [inline dapp/src/dapp/components/login.cljs:31]: Let's follow the pattern that @crimeminister suggested. 

I'm even fond of something like
`{:error/message "some useful message" :error/stacktrace error}`

Not too particular about the names but happy to hear @crimeminister's thoughts

File: dapp/src/dapp/pages/dashboard.cljs [hunk 3]
```diff
@@ -29,7 +30,7 @@
                       (resolve digest)))))))))
 
 (defn open-modal []
-  (prn "open the modal")
+  (log/trace "open the modal")
```
Comment (drew-patel) [inline dapp/src/dapp/pages/dashboard.cljs:33]: Wrap in hash-map? I think we can do an error template function and we can just use that everywhere for each type of error down the line. Don't have to do that now.

File: dapp/src/dapp/wallet.cljs [hunk 4]
```diff
@@ -40,46 +37,38 @@
...
       (re-frame/dispatch [::connect-account ctx]))))
 
 
 ;Handle a connection to different wallets and kick off the zk-auth
 (re-frame/reg-event-db ::connect-account
   (fn [db [_ wallet]]
-    (prn {:wallet wallet})
     (let [web3 ^js/Web3 (:web3 db)
-          _ (prn {:web3 web3 :db db})
+          _ (log/debug {:msg "connect-account" :wallet wallet :web3 web3 :db db})
```
Comment (drew-patel) [inline dapp/src/dapp/wallet.cljs:63]: Delete `:db db` here? Maybe too much to log. Or just log the `:sdk/ctx` from the `db`?

**Review by drew-patel — CHANGES_REQUESTED** [review]
Minor stuff. Feel free to push back if you feel it's too early to implement some of this stuff

**Review by drew-patel — APPROVED** [review]

**Pull request merged**